### ISSUE TITLE
Allow not specifying the key for link patterns

### ISSUE THREAD

**Issue by mckornfield** [jaegertracing/jaeger-ui#746]
## Requirement - what kind of business use case are you trying to solve?

* Allowing a user to configure a link pattern for all links patterns of a certain type to allow easy filtering, e.g. for all link patterns of type tag, add a quick filter:
```
{
          "type": "tag",
          "url": "https://my-jaeger.com/search?tags=%7B%#{key}%22%3A%#{value}%22%7D",
          "text": "Filter by: #{key}"
}
```

## Problem - what in Jaeger blocks you from solving the requirement?

The fact that the key field is required for Link Patterns

## Proposal - what do you suggest to solve the problem or improve the existing situation?

Allowing the key field to be optional, or maybe a boolean `applyForAllOfType: true`, that instead replaces the key with a reference to the name `key` and its value accessible via `value`

**Comment by yurishkuro** [comment]
Is the search via Jaeger UI by the given tag the intended use case? Is that the only use case? Because then it might be better to hardcode this, e.g. show a magnifying class icon on hover and leave the link patterns for their original deep linking purposes.

**Comment by albertteoh** [comment]
cc @yoave23

**Comment by mckornfield** [comment]
That's the main usecase @yurishkuro , agreed hard-coding might be a better solution. One other maybe useful thing would be if you wanted to also cross reference with another tag, like a process tag while templating (not sure if that's possible now)

**Comment by jkowall** [comment]
@yurishkuro does #817 fix this issue? If so, we can close.

**Comment by yurishkuro** [comment]
We should ask the OP, @mckornfield

**Comment by mckornfield** [comment]
Yep I think it does, I'll close fixed by #817
